src/sparseml/exporters/transforms/utils/helpers.py

```diff
@@ -161,3 +161,115 @@ def assert_node_type(node: NodeProto, op: Union[List[str], Set[str], str]) -> bo
         return node.op_type == op
     else:
         return node.op_type in op
+
+
+NodeOrInit = Union[NodeProto, TensorProto]
+
+
+class MatchResult:
+    def __init__(self, node: NodeOrInit) -> None:
+        self.node: Optional[NodeOrInit] = node
+        self.parents: List[Optional[NodeOrInit]] = []
+        self.children: List[Optional[NodeOrInit]] = []
+
+
+INITIALIZER_MATCH = "__Initializer__"
+
+
+def optional_node(tag: str) -> str:
+    return "Optional-" + tag
+
+
+def _is_optional_node(tag: str) -> Tuple[str, bool]:
+    if tag.startswith("Optional-"):
+        return tag.split("-")[1], True
+    else:
+        return tag, False
+
+
+def iter_structural_matches(
```

Comment:
I'd add docstrings to the public functions in this script, they would be potentially:
- hard for us to decipher if we return to working on them in the future
- hard for somebody to expand on our work

I am very grateful to Tuan for leaving bits and pieces of comments in his code, that we are now refactoring. Let's employ the same mindset.

Reply:
Good call out!